File: src/main/java/upbrella/be/store/dto/response/StoreIntroductionsResponseByClassification.java
package upbrella.be.store.dto.response;

import lombok.Builder;
import lombok.Getter;
import upbrella.be.store.entity.StoreDetail;

import java.util.List;
import java.util.stream.Collectors;

@Getter
@Builder
public class StoreIntroductionsResponseByClassification {

    private long subClassificationId;
    private List<SingleStoreIntroductionResponse> stores;

    public static StoreIntroductionsResponseByClassification of(long subClassificationId, List<StoreDetail> storeDetails) {

        return StoreIntroductionsResponseByClassification.builder()
                .subClassificationId(subClassificationId)
                .stores(storeDetails.stream()
                                .map(SingleStoreIntroductionResponse::createSingleIntroduction)
                                .collect(Collectors.toList()))
                .build();
    }
}

File: src/main/kotlin/upbrella/be/store/dto/response/StoreIntroductionsResponseByClassification.kt
package upbrella.be.store.dto.response

import upbrella.be.store.entity.StoreDetail

data class StoreIntroductionsResponseByClassification(
    val subClassificationId: Long,
    val stores: List<SingleStoreIntroductionResponse>
) {
    companion object {
        fun of(subClassificationId: Long, storeDetails: List<StoreDetail>): StoreIntroductionsResponseByClassification {
            return StoreIntroductionsResponseByClassification(
                subClassificationId = subClassificationId,
                stores = storeDetails.map { SingleStoreIntroductionResponse.createSingleIntroduction(it) }
            )
        }
    }
}
